crates/sui-source-validation-service/src/lib.rs

```diff
@@ -438,6 +439,8 @@ pub async fn watch_for_upgrades(
                     channel.send(result).unwrap();
                     break Ok(());
                 }
+                info!("Shutting down server (resync performed on restart)");
+                std::process::exit(1)
```

Comment:
With this, our k8s infra will kick the server on protocol upgrade (rather than needing to manually detect / kick pods).